src/components/Photo/DocumentLiveCapture.js

```diff
@@ -0,0 +1,121 @@
+// @flow
+import * as React from 'react'
+import { h, Component } from 'preact'
+import { screenshot } from '~utils/camera.js'
+import { mimeType } from '~utils/blob.js'
+import { DocumentOverlay } from '../Overlay'
+import { ToggleFullScreen } from '../FullScreen'
+import Spinner from '../Spinner'
+import Timeout from '../Timeout'
+import Camera from '../Camera'
+import CameraError from '../CameraError'
+import style from './style.css'
+
+type State = {
+  hasBecomeInactive: boolean,
+  hasCameraError: boolean,
+  isLoading: boolean
+}
+
+type Props = {
+  translate: (string, ?{}) => string,
+  onCapture: Function,
+  renderFallback: Function,
+  isUploadFallbackDisabled: boolean,
+  trackScreen: Function,
+  documentType: string,
+  className: string,
+  containerClassName: string,
+  renderTitle: Function,
+  renderError: Function
+}
+
+export default class DocumentLiveCapture extends Component<Props, State> {
+  webcam = null
+
+  state: State = {
+    hasBecomeInactive: false,
+    hasCameraError: false,
+    isLoading: false
+  }
+
+  handleTimeout = () => this.setState({ hasBecomeInactive: true })
+
+  handleCameraError = () => this.setState({ hasCameraError: true })
+
+  inactiveError = () => ({
+    name: this.props.isUploadFallbackDisabled ? 'CAMERA_INACTIVE_NO_FALLBACK' : 'CAMERA_INACTIVE',
+    type: 'warning'
+  })
+
+  captureDocument = (blob: Blob, sdkMetadata: Object) => {
+    const documentCapture = {
+      blob,
+      sdkMetadata,
+      filename: `document_capture.${mimeType(blob)}`
+    }
+    this.props.onCapture(documentCapture)
+  }
+
+  captureDocumentPhoto = () => {
+    this.setState({ isLoading: true })
```

Comment:
Don't you have to unset `isLoading` in case of an error?

Reply:
There is actually no service call happening here, the loading state is just to compensate for a slight delay of about 1 second when transitioning between the capture screen and the "confirm image" screen on lower end mobiles. 

But I'll add code to unset `isLoading` on `componentWillUnmount` anyway.